docs/blog/2025-09-16-creating-an-nx-plugin-for-biome.md

```diff
@@ -0,0 +1,448 @@
+
+## Using npm scripts
+
+Normally, to avoid writing long commands every time, we'd reach for npm scripts. Let's add this to our root `package.json`:
+
+```json {% fileName="package.json" %}
+{
+  "scripts": {
+    "biome-lint": "biome lint"
+  }
+}
+```
+
+Now we can run `npm run biome-lint` instead. But we're still linting the entire workspace.
+
+### Targeting Individual Projects
+
+Since our workspace uses npm workspaces, every project has its own `package.json`. We can add the same script to individual project package files:
+
+```json {% fileName="apps/biome-example/package.json" %}
+{
+  "scripts": {
+    "biome-lint": "biome lint"
+  }
+}
+```
+
+Now if we move into that directory and run `npm run biome-lint`, we're only linting that specific directory. When you run `biome lint` with no other options, it lints that directory and directories below it.
+
+## The Nx Integration you didn't realize was there
+
+Here's something you might not realize: **we've already started an Nx integration**. If you open Nx Console, you'll see that `biome-lint` is available as a target under the "npm scripts" area. Nx automatically incorporates npm scripts into available tasks.
+
+![Screenshot of Nx Console showing npm scripts available as targets](/blog/images/2025-09-16/npm-scripts.avif)
+
+So while `npm run biome-lint` works, we can also run this through Nx:
+
+```shell
+npx nx biome-lint biome-example
+```
+
+This gives us the exact same result. We can copy this script to other packages, like our navigation library, and suddenly we can run `biome-lint` on multiple projects individually.
+
+## Hold on, I'm not using npm workspaces!
+
+While npm workspaces are the default for new Nx workspaces, there's a long history of workspaces created before that. Those workspaces use `project.json` files for project configuration. While you won't be able to use npm scripts like in npm workspaces, you can still create a target in your `project.json` files:
+
+```json {% fileName="apps/biome-example/project.json" %}
+{
+  "targets": {
+    "biome-lint": {
+      "command": "npx @biomejs/biome lint {projectRoot}"
+    }
+  }
+}
+```
+
+You'll notice one small difference: `{projectRoot}` appended to the end of the command. We'll explain this part when we're creating our inferred task plugin later, because this configuration is _exactly_ what you'll be creating as part of the plugin.
+
+## Adding Caching Support
+
+One thing missing with this simple setup is **caching**. Nx doesn't know the `biome-lint` command is cacheable, and it doesn't know what to cache.
+
+Right now, if we rerun `nx biome-lint biome-example` twice, it actually runs the lint every time. Ideally, we shouldn't rerun this lint if we haven't changed anything between successful runs.
+
+### Configuring Target Defaults
```

Comment:
I would just remove this subsection and keep it in the adding caching support section. Basically just have a sentence that says to configure caching, open your Nx JSON and add a target defaults, etc.